### PR TITLE
Blob metadata

### PR DESCRIPTION
**Pull request by Lucretius**
Resolves https://github.com/terraform-providers/terraform-provider-azurerm/issues/1400

- Added support for `metadata` property for storage blobs
- Updated markdown by adding property
- Added metadata to basic test (if desired, I can split this into a *separate* test for metadata, but seeing as metadata is like a tag it looks like we normally just include those in basic tests)

### CODE REVIEW

File: azurerm/resource_arm_storage_blob.go [hunk 1]
```diff
@@ -201,6 +209,13 @@ func resourceArmStorageBlobCreate(d *schema.ResourceData, meta interface{}) erro
 		}
 	}
 
+	blob.Metadata = expandBlobMetadata(d)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_blob.go]: As these functions are package wide could we rename this to
```suggestion
	blob.Metadata = expandStorageAccountBlobMetadata(d)
```

File: azurerm/resource_arm_storage_blob.go [hunk 2]
```diff
@@ -564,6 +579,15 @@ func resourceArmStorageBlobUpdate(d *schema.ResourceData, meta interface{}) erro
 		return fmt.Errorf("Error setting properties of blob %s (container %s, storage account %s): %+v", id.blobName, id.containerName, id.storageAccountName, err)
 	}
 
+	if d.HasChange("metadata") {
+		blob.Metadata = expandBlobMetadata(d)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_blob.go]: As these functions are package wide could we rename this to
```suggestion
		blob.Metadata = expandStorageAccountBlobMetadata(d)
```

File: azurerm/resource_arm_storage_blob.go [hunk 3]
```diff
@@ -632,6 +662,7 @@ func resourceArmStorageBlobRead(d *schema.ResourceData, meta interface{}) error
 		log.Printf("[INFO] URL for %q is empty", id.blobName)
 	}
 	d.Set("url", u)
+	d.Set("metadata", flattenBlobMetadata(blob.Metadata))
```
Comment (katbyte) [inline azurerm/resource_arm_storage_blob.go]: and here:
```suggestion
	d.Set("metadata", flattenStorageAccountBlobMetadata(blob.Metadata))
```

File: azurerm/resource_arm_storage_blob.go [hunk 4]
```diff
@@ -114,6 +114,14 @@ func resourceArmStorageBlob() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validation.IntAtLeast(1),
 			},
+
+			"metadata": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
```
Comment (katbyte) [inline azurerm/resource_arm_storage_blob.go]: Should we validate this with `NoEmptyStrings`?

Comment (Lucretius) [reply azurerm/resource_arm_storage_blob.go]: I think that is a good call

**Comment by Lucretius** [comment]
Thanks @katbyte for the feedback, I've renamed the functions as desired.

**Comment by bpoland** [comment]
Thanks everyone!

**Pull request merged**